lib/queue.rb

```diff
@@ -1,31 +1,72 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(50)
+    @front = -1
+    @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if((@front == 0 && @back == @store.length - 1) || (@front == @back + 1))
+      raise ArgumentError
+    elsif @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+    
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
   end
-
+  
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1
+      raise ArgumentError
+    end
+    
+    data = @store[@front]
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+    
+    if @front == @back
+      @front = -1
+      @back = -1
+    end
+    
+    return data
   end
-
+  
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
-
+  
   def size
-    raise NotImplementedError, "Not yet implemented"
+    if self.empty?
+      raise ArgumentError
+    end
+    
+    return @back - @front
```

Comment:
This only works if @front < @back,

```suggestion
    return @back - @front if @front < @back
    return @back + @store.length - @front
```